### PR TITLE
feat: implement (basic) webidl & headers comments

### LINKED ISSUE #933: Add spec comments to Header class
It is useful to add comment references to the relevant parts of the whatwg fetch spec for the Headers class.

### PR DESCRIPTION
**Pull request by KhafraDev**
Fixes #933 

WebIdl is pretty cool and it's used rather heavily in the fetch spec. Implementing a subset of it (that we need) will make everything much easier once it's finished. I can explain how it'll help if you want, just don't want to write out an essay if you don't care lol

I only did the `Headers` class as a PoC and in the process discovered a *load* of bugs and decided just to add the spec comments since I was already adding a ton to it and fixing stuff anyways 😄.

p.s.: it's very clear that `HeadersList` was not supposed to be Map-like structure, as each key can be duplicated and the actual appending doesn't occur until you use `.get(...)`. Not a big deal, just something I noticed when working on it.

### CODE REVIEW

**Review by ronag — APPROVED** [review]

File: lib/fetch/webidl.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const { toUSVString } = require('util')
```
Comment (mcollina) [inline lib/fetch/webidl.js:1]:
```suggestion
'use strict'

const { toUSVString } = require('util')
```

**Review by mcollina — CHANGES_REQUESTED** [review]
Could you add some tests for the new code?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by mcollina — CHANGES_REQUESTED** [review]
Could you add some tests for the new code?

**Comment by KhafraDev** [comment]
added tests

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**